wlroots/wlr_types/output_damage.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Matt Colligan 2021
+
+from pywayland.server import Signal
+
+from wlroots import ffi, lib, Ptr
+from wlroots.util.region import PixmanRegion32
+from .box import Box
+from .output import Output
+
+
+class OutputDamage(Ptr):
+    def __init__(self, ptr) -> None:
+        """
+        Tracks damage for an output.
+
+        The `frame` event will be emitted when it is a good time for the compositor
+        to submit a new frame.
+
+        To render a new frame, compositors should call
+        `wlr_output_damage_attach_render`, render and call `wlr_output_commit`. No
+        rendering should happen outside a `frame` event handler or before
+        `wlr_output_damage_attach_render`.
+
+        :param ptr:
+            The wlr_output_damage cdata pointer
+        """
+        self._ptr = ffi.cast("struct wlr_output_damage *", ptr)
+
+        self.frame_event = Signal(ptr=ffi.addressof(self._ptr.events.frame))
+        self.destroy_event = Signal(ptr=ffi.addressof(self._ptr.events.destroy))
+
+    @classmethod
+    def create(cls, output: Output) -> "OutputDamage":
+        """Create an instance for the given wlr_output"""
+        ptr = lib.wlr_output_damage_create(output._ptr)
+        return OutputDamage(ptr)
+
+    @property
+    def output(self) -> Output:
+        """The name of the output"""
+        return Output(self._ptr.output)
+
+    def destroy(self) -> None:
+        """The name of the output"""
+        lib.wlr_output_damage_destroy(self._ptr)
```

Comment:
Should this be set as the `gc` action of `self._ptr`?

Reply:
What implications does that have? Does the `OutputDamage` have to maintain a reference or be destroyed? (Not sure why you'd have an `OutputDamage` instantiated and _not_ keep a reference around tho).

Reply:
Ah, right, if the output damage object is deleted because the reference is not kept alive, then it would be automatically destroyed. I'm not sure if there are problems with that causing them to be destroyed too early, or if there would be problems with not coming back and destroying them all explicitly.